### PR TITLE
allow more flexible plugin test files `test.*.py`

### PR DESCRIPTION
**Pull request by mschrimpf**

### CODE REVIEW

File: brainscore_core/plugin_management/test_plugins.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+import re
 import warnings
 from pathlib import Path
 from typing import Dict, Union
 
-import pytest_check as check
 from .environment_manager import EnvironmentManager
 
 PLUGIN_TYPES = ['benchmarks', 'data', 'metrics', 'models']
+RECOGNIZED_TEST_FILES = 'test.*.py'
```
Comment (kvfairchild) [inline brainscore_core/plugin_management/test_plugins.py:10]:
```suggestion
RECOGNIZED_TEST_FILES = 'test.*\.py'
```

File: brainscore_core/plugin_management/test_plugins.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
 
 class PluginTestRunner(EnvironmentManager):
-    """Runs plugin tests (requires "test.py" for each plugin)
+    """Runs plugin tests (requires "test.*.py" for each plugin)
```
Comment (kvfairchild) [inline brainscore_core/plugin_management/test_plugins.py:14]:
```suggestion
    """Runs plugin tests (requires "test.*\.py" for each plugin)
```

File: brainscore_core/plugin_management/test_plugins.py [hunk 3]
```diff
@@ -43,8 +45,10 @@ def __call__(self):
         self.teardown()
 
     def validate_plugin(self):
-        """ requires "test.py" file in plugin directory """
-        assert (self.plugin_directory / 'test.py').is_file(), "'test.py' not found"
+        """ requires at least one file matching "test.*.py" in plugin directory, e.g. test.py, test_data.py. """
```
Comment (kvfairchild) [inline brainscore_core/plugin_management/test_plugins.py:48]:
```suggestion
        """ requires at least one file matching "test.*\.py" in plugin directory, e.g. test.py, test_data.py. """
```

File: brainscore_core/plugin_management/test_plugins.py [hunk 4]
```diff
@@ -43,8 +45,10 @@ def __call__(self):
...
+        test_files = [test_file for test_file in self.plugin_directory.iterdir()
+                      if re.match(RECOGNIZED_TEST_FILES, test_file.name)]
+        assert len(test_files) > 0, "No test files matching 'test.*.py' found"
```
Comment (kvfairchild) [inline brainscore_core/plugin_management/test_plugins.py:51]:
```suggestion
        assert len(test_files) > 0, "No test files matching 'test.*\.py' found"
```

File: tests/test_plugin_management/test_test_plugins.py [hunk 5]
```diff
@@ -23,13 +23,25 @@ def test_plugin_name(self):
...
+        test_file = self.dummy_plugin_path / 'test.py'
+        test_file.rename(self.dummy_plugin_path / 'test_plugin.py')
+        plugin_test_runner = PluginTestRunner(self.dummy_plugin_path, {})
+        plugin_test_runner.validate_plugin()
+
+    def test_has_testfile_no_underscore_prefix(self):
+        test_file = self.dummy_plugin_path / 'test.py'
+        test_file.rename(self.dummy_plugin_path / 'testplugin.py')
+        plugin_test_runner = PluginTestRunner(self.dummy_plugin_path, {})
+        plugin_test_runner.validate_plugin()
```
Comment (kvfairchild) [inline tests/test_plugin_management/test_test_plugins.py:43]:
```suggestion
        plugin_test_runner.validate_plugin()
        
        def test_has_testfile_valid_python(self):
        test_file = self.dummy_plugin_path / 'test.py'
        test_file.rename(self.dummy_plugin_path / 'testpluginpy')
        plugin_test_runner = PluginTestRunner(self.dummy_plugin_path, {})
        with pytest.raises(Exception):
            plugin_test_runner.validate_plugin()
```

File: brainscore_core/plugin_management/test_plugins.py (hunk 1 above)
Comment (kvfairchild) [reply brainscore_core/plugin_management/test_plugins.py:10]: (Second dot needs escaping to prevent files named `testpy`)

**Review by kvfairchild — CHANGES_REQUESTED** [review]

**Comment by mschrimpf** [comment]
@kvfairchild ping

**Review by kvfairchild — APPROVED** [review]

**Pull request merged**